azurerm/internal/services/network/parse/express_route_circuit_connection.go

```diff
@@ -0,0 +1,56 @@
+package parse
+
+import (
+	"fmt"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+)
+
+type NetworkExpressRouteCircuitConnectionId struct {
+	SubscriptionId string
+	ResourceGroup  string
+	CircuitName    string
+	PeeringName    string
+	Name           string
+}
+
+func NewExpressRouteCircuitConnectionID(subscriptionId string, resourcegroup string, circuitname string, peeringname string, name string) NetworkExpressRouteCircuitConnectionId {
+	return NetworkExpressRouteCircuitConnectionId{
+		SubscriptionId: subscriptionId,
+		ResourceGroup:  resourcegroup,
+		CircuitName:    circuitname,
+		PeeringName:    peeringname,
+		Name:           name,
+	}
+}
+
+func (id NetworkExpressRouteCircuitConnectionId) ID() string {
+	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/expressRouteCircuits/%s/peerings/%s/connections/%s"
```

Comment:
these should be generated using `go:generate`: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/network/resourceids.go#L4

Reply:
Updated